oktoberfest/predict/predict.py

```diff
@@ -134,11 +137,10 @@ def _alignment(alignment_library: Spectra):
 
     :param alignment_library: the library to perform the alignment on
     """
-    pred_intensity = alignment_library.get_matrix(FragmentType.PRED)[0]
-    raw_intensity = alignment_library.get_matrix(FragmentType.RAW)[0]
-    # return pred_intensity.toarray(), raw_intensity.toarray()
+    pred_intensity = alignment_library.get_matrix(FragmentType.PRED)[0].toarray()
+    raw_intensity = alignment_library.get_matrix(FragmentType.RAW)[0].toarray()
     sm = SimilarityMetrics(pred_intensity, raw_intensity)
-    alignment_library.spectra_data["SPECTRAL_ANGLE"] = sm.spectral_angle(raw_intensity, pred_intensity, 0)
+    alignment_library.add_column(sm.spectral_angle(raw_intensity, pred_intensity, 0), "SPECTRAL_ANGLE")
     alignment_library.spectra_data = alignment_library.spectra_data[
-        alignment_library.spectra_data["SPECTRAL_ANGLE"] != 0
+        alignment_library.spectra_data.obs["SPECTRAL_ANGLE"] != 0, :
```

Comment:
you don't need the ":", simply indexing on the first axis is fine, since this is boolean.